components/StakingPageWrapper.tsx

```diff
@@ -20,8 +20,9 @@ type StakingPageWrapperProps = {
 
 function StakingPageWrapper({ children, showBg }: StakingPageWrapperProps) {
   const { showAlert } = useAlert()
-  const isDesktop = useRecoilValue(isDesktopState)
-  const setInvalidPrice = useSetRecoilState(invalidPriceState)
+
+  const isDesktop = useAtomValue(isDesktopAtom)
+  const setInvalidPrice = useSetAtom(invalidPriceAtom)
```

Comment:
I think this value can be replaced with the value directly from the `useQuery` call below. Maybe make a custom hook for this:

```typescript
function useInvalidPriceValue() {
	return useQuery(['fallbackTokenPrices'], fetchCoinmarketCapTokenPrice, {
	    retry: false,
	    staleTime: Infinity,
	    keepPreviousData: true,
	    placeholderData: {},
	}).isError;
}
```

I think this would be beneficial because:
- Reduces a rerender (probably, untested)
- No concerns about possible data mismatch between the query and the atom value.

Deriving state wherever possible is quite a good practice to follow: https://kentcdodds.com/blog/dont-sync-state-derive-it

Reply:
One thing that concerns me is that we can't instantly know whether `invalidPriceError` is thrown unless we run `useInvalidPriceValue`. (Don't we?)

**So, my suggestions is:**

1. Delete `invalidPriceError` state
2. Derive `invalidPriceError` from `useQueryResult`

```ts
export function usePrices() {
  const fallbackTokenPrices = useQuery(
    ['fallbackTokenPrices'],
    fetchCoinmarketCapTokenPrice,
    { ... }
  )

  const tokenPrices = useQuery<TokenPrices>(
    ['tokenPrices', addresses],
    () => fetchTokenPrices(addresses),
    {
      ...,
      onError() {
        const state = queryClient.getQueryData<TokenPrices>(['fallbackTokenPrices']) || TOKEN_PRICES_PLACEHOLDERS
        queryClient.setQueryData(['tokenPrices', addresses], state)
      },
    }
  )

  const invalidPriceError = useMemo(
    () => fallbackTokenPrices.isError && tokenPrices.isError,
    [fallbackTokenPrices.isError, tokenPrices.isError]
  )

  return {
    ...,
    invalidPriceError
  }
}
```

Reply:
Nice trick!